File: Python/String-2/count_hi.py

# return the number of times that the string "hi" appears anywhere in the given string
def count_hi(str):
    count = 0

    for x in range(len(str)-1):
        if str[x] == 'h' and str[x+1] == 'i':
            count += 1

    return count

print(count_hi('abc hi ho'))
print(count_hi('ABChi hi'))
print(count_hi('hihi'))

File: Java/String-2/countHi.java

public class string2 {
	
	// return the number of times that the string "hi" appears anywhere in the given string
	public int countHi(String str) {
		
		int count = 0;
		  
		for (int i = 0; i < str.length() - 1; i++) {
			if (str.charAt(i) == 'h' && str.charAt(i+1) == 'i')
				count++;
		}  
		  
		return count;
		
	}

	public static void main(String[] args) {

		string2 o = new string2();
		
		System.out.println(o.countHi("abc hi ho"));
		System.out.println(o.countHi("ABChi hi"));
		System.out.println(o.countHi("hihi"));

	}

}
